### ISSUE TITLE
String overflow error in GCC 11 

### ISSUE THREAD

**Issue by disa6302** [Mbed-TLS/mbedtls#8204]
### Summary

When building this with GCC 11 I see:

```
error: ‘mbedtls_sha512_finish_ret’ accessing 64 bytes in a region of size 48 [-Werror=stringop-overflow=]
 3267 |     finish( &sha512, padbuf );
 ```

### System information

Mbed TLS version (number or commit id): v2.25
Operating system and version: Ubuntu 22.04
Configuration (if not default, please attach `mbedtls_config.h`): 
Compiler and options (if you used a pre-built binary, please indicate how you obtained it): 
Additional environment information: 

I am using this project as an external project in my project:

```
cmake_minimum_required(VERSION 3.6.3)

project(libmbedtls-download NONE)

include(ExternalProject)

if (BUILD_STATIC_LIBS)
  set(BUILD_SHARED 0)
else()
  set(BUILD_SHARED 1)
endif()

if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
  set(CMAKE_MACOSX_RPATH 1)
else()
  set(CMAKE_MACOSX_RPATH 0)
endif()

ExternalProject_Add(
  project_libmbedtls
  GIT_REPOSITORY  https://github.com/ARMmbed/mbedtls.git
  GIT_TAG         v2.25.0
  PREFIX          ${CMAKE_CURRENT_BINARY_DIR}/build
  CMAKE_ARGS
    -DCMAKE_INSTALL_PREFIX=${OPEN_SRC_INSTALL_PREFIX}
    -DUSE_SHARED_MBEDTLS_LIBRARY=${BUILD_SHARED}
    -DCMAKE_BUILD_TYPE=Release
    -DCMAKE_MACOSX_RPATH=${CMAKE_MACOSX_RPATH}
    -DENABLE_TESTING=OFF
    -DENABLE_PROGRAMS=OFF
    -DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}
  BUILD_ALWAYS    TRUE
  TEST_COMMAND    "")
```

### Expected behavior

Builds clean

### Actual behavior

Fails with the error

### Steps to reproduce

Include this as a depedency or build as such

### Additional information

I went through multiple issues and noticed that the fix was backported to 2.x. However, I am not sure if this includes v2.25 because it does not seem to build.

**Comment by daverodgman** [comment]
As you note, this was fixed by #4493. 2.25 is not an LTS release, and will not receive further fixes.

I recommend upgrading to the latest 2.28 LTS release (currently https://github.com/Mbed-TLS/mbedtls/releases/tag/v2.28.4, although 2.28.5 will be released within the next few weeks). 2.28 LTS is supported until end of 2024. Alternatively, you could move to the latest release (3.4, soon to be 3.5, with 3.6 LTS to be released around end of year, or early 2024), but this might be a more disruptive upgrade than 2.25 to 2.28.

**Comment by disa6302** [comment]
Got it. Thank you for your response!
